src/app/portlets/content-types/fields/content-type-fields-add-row/content-type-fields-add-row.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { ContentTypeFieldsAddRowComponent } from './content-type-fields-add-row.component';
+import { MessageService } from './../../../../api/services/messages-service';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { TooltipModule } from 'primeng/primeng';
+import { async, ComponentFixture, fakeAsync, tick } from '@angular/core/testing';
+import { DOTTestBed } from '../../../../test/dot-test-bed';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { Field } from '../';
+import { Observable } from 'rxjs/Observable';
+import { HotkeysService, Hotkey } from 'angular2-hotkeys';
+import { TestHotkeysMock } from './../../../../test/hotkeys-service.mock';
+import { MockMessageService } from './../../../../test/message-service.mock';
+
+describe('ContentTypeFieldsAddRowComponent', () => {
+    let comp: ContentTypeFieldsAddRowComponent;
+    let fixture: ComponentFixture<ContentTypeFieldsAddRowComponent>;
+    let de: DebugElement;
+    let el: HTMLElement;
+    let testHotKeysMock: TestHotkeysMock;
+
+    const messageServiceMock = new MockMessageService({
+        'contenttypes.content.add_rows': 'Add Rows'
+    });
+
+    beforeEach(async(() => {
+        testHotKeysMock = new TestHotkeysMock();
+
+        DOTTestBed.configureTestingModule({
+            declarations: [
+                ContentTypeFieldsAddRowComponent
+            ],
+            imports: [
+                TooltipModule,
+                BrowserAnimationsModule
+            ],
+            providers: [
+                { provide: HotkeysService, useValue: testHotKeysMock },
+                { provide: MessageService, useValue: messageServiceMock }
+            ]
+        });
+
+        fixture = DOTTestBed.createComponent(ContentTypeFieldsAddRowComponent);
+        comp = fixture.componentInstance;
+        de = fixture.debugElement;
+        el = de.nativeElement;
+    }));
+
+    it('should display the add rows button by default', () => {
+        const addRowContainer = de.nativeElement.querySelector('.dot-add-rows__container');
+
+        fixture.detectChanges();
+
+        expect(addRowContainer.classList.contains('dot-add-rows__add')).toEqual(true);
+    });
+
+    it('should display a row after click on AddRows button', () => {
```

Comment:
This test will pass without the ```.click``` because even when the list it's "hidden" for the position the selector of the ```<li>``` will still be able to grab them. For this case it's better to check if the class change or the position of the top.

**NOTE:** When you are doing tests is a good practice to make them fail and them fix them to make them pass.